### PR TITLE
Unify cached schema use between topics and schemas (issue 214)

### PR DESCRIPTION
**Pull request by jlrobins**
## Summary of Changes

- Unify cached schema use between topics and schemas view providers.
- Make the preloader keep track of 'coarse' vs 'fine grained' resources, with 'coarse' being the environments, kafka clusters, and schema registries. Schemas are now fine-grained cached (and refresh-able)  per schema registry cluster id. Stop pre-loading all of the schemas at ccloud log in time, but instead do it on demand whenever either any topics are loaded for a cluster, or the schema registry is selected and we need to show all of the schemas.
- Fix premature clearing of the deepRefresh flags: ensure it gets cleared after the refreshing is completed, not right after the repaint event fires (but probably before the flag has been observed).
- Simplify the public schema-related methods from ResourceManager, removing un-called methods and their tests.

## Any additional details or context that should be provided?

- When selecting a schema registry to view for the second time, or after when one of its peer kafka clusters has had its topics loaded, then displaying them a subsequent time will draw from cache.
- Completing the deep fetch for the `Resources` panel is now faster, given that it no longer deep fetches all of the schemas from all of the reachable ccloud schema registries.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [x] Added new
- [x] Updated existing
- [x] Deleted existing

##### Other

- [x] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [x] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ All notable changes to this extension will be documented in this file.
 
 - Use cached information to populate the Resources view Confluent Cloud single environment children,
   [issue #254](https://github.com/confluentinc/vscode/issues/254).
+- Integrate and synchronize use of cached schemas information between Topics and Schemas panels,
```
Comment (shouples) [inline CHANGELOG.md:11]:
```suggestion
- Integrate and synchronize use of cached schemas information between Topics and Schemas views,
```

**Comment by jlrobins** [comment]
Have two tests to fix due to resources manager cleanup, will do tomorrow:
```

  2 failing
  1) TopicViewProvider helper functions
       loadTopicSchemas() should return schemas for CCloud Kafka topics when available:
     TypeError: resourceManager$1.setCCloudSchemas is not a function
  	at Context.<anonymous> (out/src/viewProviders/topics.test.js:90:29)

  2) TopicViewProvider helper functions
       loadTopicSchemas() should not return schemas for CCloud Kafka topics if none are available in extension state:
     TypeError: resourceManager.getResourceManager(...).setCCloudSchemas is not a function
  	at Context.<anonymous> (out/src/viewProviders/topics.test.js:98:48)
```

File: src/viewProviders/topics.ts [hunk 2]
```diff
@@ -173,14 +186,21 @@ export async function getTopicsForCluster(
...
 
   if (cluster instanceof CCloudKafkaCluster) {
     environmentId = cluster.environmentId;
 
     const schemaRegistry = await resourceManager.getCCloudSchemaRegistryCluster(environmentId);
     if (schemaRegistry) {
-      schemas = await resourceManager.getCCloudSchemasForCluster(schemaRegistry.id);
+      schemas = await resourceManager.getSchemasForRegistry(schemaRegistry.id);
+      if (schemas === undefined) {
+        logger.error("Wacky: schema registry known, but unknown schemas (should be empty array)", {
```
Comment (noeldevelops) [inline src/viewProviders/topics.ts:198]: 😆

**Comment by jlrobins** [comment]
Fixed tests and lint / check warts. Passes 'gulp test check lint.'

**Review by shouples — APPROVED** [review]
Thanks for putting this together! I think the experience is a lot nicer -- faster, fewer edge cases, and the split in preloading resources is easier to understand as well on the extension development side. Awesome stuff. 🎉

File: src/viewProviders/resources.ts [hunk 3]
```diff
@@ -109,7 +108,12 @@ export class ResourceViewProvider implements vscode.TreeDataProvider<ResourceVie
     } else {
       // --- ROOT-LEVEL ITEMS ---
       // NOTE: we end up here when the tree is first loaded
-      return await loadResources(this.forceDeepRefresh);
+      const toplevelItems = await loadResources(this.forceDeepRefresh);
+      if (this.forceDeepRefresh) {
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/viewProviders/resources.ts:112]: Damned if do, damned if don't. Half the people like style of 'always clear the flag even if wasn't set in first place' in argument of 'is faster and fewer LOC', but other half like the explicitness. I erred on the latter this time given isn't anywhere close to a high performance inner loop needing to avoid branch prediction misses, and also gives a good place for the comment tying off the deep refresh flow. YMMV, OC.
